module/VuFind/src/VuFind/View/Helper/Root/CleanHtmlFactory.php

```diff
@@ -0,0 +1,147 @@
+
+    /**
+     * Create an object
+     *
+     * @param ContainerInterface $container     Service manager
+     * @param string             $requestedName Service being created
+     * @param null|array         $options       Extra options (optional)
+     *
+     * @return object
+     *
+     * @throws ServiceNotFoundException if unable to resolve the service.
+     * @throws ServiceNotCreatedException if an exception is raised when
+     * creating a service.
+     * @throws ContainerException if any other error occurs
+     */
+    public function __invoke(
+        ContainerInterface $container,
+        $requestedName,
+        array $options = null
+    ) {
+        if (!empty($options)) {
+            throw new \Exception('Unexpected options sent to factory.');
+        }
+
+        $this->container = $container;
+        return new $requestedName(Closure::fromCallable([$this, 'createPurifier']));
+    }
+
+    /**
+     * Create a purifier instance.
+     *
+     * N.B. This is a relatively slow method.
+     *
+     * @param array $options Additional options. Currently supported:
+     *   targetBlank  true/false  Whether to add target="_blank" to external links
+     *
+     * @return HTMLPurifier
+     */
+    protected function createPurifier(array $options): HTMLPurifier
+    {
+        $config = \HTMLPurifier_Config::createDefault();
+        // Set cache path to the object cache
+        $cacheDir
+            = $this->container->get(\VuFind\Cache\Manager::class)->getCache('object')->getOptions()->getCacheDir();
+        if ($cacheDir) {
+            $config->set('Cache.SerializerPath', $cacheDir);
+        }
+        if ($options['targetBlank'] ?? false) {
+            $config->set('HTML.Nofollow', 1);
+            $config->set('HTML.TargetBlank', 1);
+        }
+
+        // Setting the following option makes purifier’s DOMLex pass the
+        // LIBXML_PARSEHUGE option to DOMDocument::loadHtml method. This in turn
+        // ensures that PHP calls htmlCtxtUseOptions (see
+        // github.com/php/php-src/blob/PHP-8.1.14/ext/dom/document.c#L1870),
+        // which ensures that the libxml2 options (namely keepBlanks) are set up
+        // properly, and whitespace nodes are preserved. This should not be an
+        // issue from libxml2 version 2.9.5, but during testing the issue was
+        // still intermittently present. Regardless of that, CentOS 7.x have an
+        // older libxml2 that exhibits the issue.
+        $config->set('Core.AllowParseManyTags', true);
+
+        $this->setAdditionalConfiguration($config);
+        return new \HTMLPurifier($config);
+    }
+
+    /**
+     * Sets additional configuration
+     *
+     * @param HTMLPurifier_Config $config Configuration
+     *
+     * @return void
+     */
+    protected function setAdditionalConfiguration(HTMLPurifier_Config $config)
+    {
+        // Add support for deprecated tags:
+        $definition = $config->getHTMLDefinition(true);
```

Comment:
I'm new to this purifier library, but reading the code and trying to see what this function has to do with deprecated tags.  I see passing `true` to `$raw` returns a pre-setup copy of the definition, but not clear how that affects deprecated tags.

Reply:
That was a bad comment. I replaced it with a new one. It's meant to describe the next few lines.